xonsh/history/json.py

```diff
@@ -422,8 +425,19 @@ def append(self, cmd):
         """
         if not self.remember_history:
             return
-        self.buffer.append(cmd)
-        self._len += 1  # must come before flushing
+
+        opts = builtins.__xonsh__.env.get("HISTCONTROL")
+        skipped_by_ignore_space = "ignorespace" in opts and cmd.get("spc")
+        if not skipped_by_ignore_space:
+            # Skipping command starting with space
+            self.buffer.append(cmd)
+            self._len += 1  # must come before flushing
+
+        try:
+            del cmd["spc"]
+        except KeyError:
+            pass
```

Comment:
Any reason we need to try to boot this key?

Reply:
I felt it was weird to leave that key around in the history file that's written out.